src/data/astarte_device_sdk_lib.rs

```diff
@@ -105,12 +106,12 @@ impl AstarteDeviceSdkConfigOptions {
             return registry
                 .read()
                 .await
-                .map_err(DeviceSdkError::ReadSecret)
-                .map(Credential::secret);
+                .map(Credential::secret)
+                .map_err(DeviceSdkError::ReadSecret);
         }
 
         if let Some(token) = &self.pairing_token {
-            return Ok(Credential::paring_token(token));
+            return Ok(Credential::secret(secret));
```

Comment:
Is it valid this changes? Shouldn't we return the token?

Reply:
Fixed